jax/random.py

```diff
@@ -98,6 +98,14 @@ def _is_prng_key(key: jnp.ndarray) -> bool:
 ### utilities
 
 
+# TODO(mattjj,jakevdp): add more info to error message, use this utility more
+def _asarray(x):
+  """A more restrictive jnp.asarray, only accepts JAX arrays and np.ndarrays."""
+  if not isinstance(x, (np.ndarray, jnp.ndarray)):
+    raise TypeError(f"Function requires array input, got {x} of type {type(x)}.")
```

Comment:
Should we accept python & numpy scalars here as well?

Reply:
I'm not sure; it might depend on how this ends up getting used. I mainly wanted to fix this particular bug (jax.random.choice not accepting ndarrays) for now, so that we could iterate on this `_asarray` method (and maybe something like it in many more places) later.

WDYT?

Reply:
Sounds good.